src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -802,9 +810,11 @@ export class CellModel extends Disposable implements ICellModel {
 		this._source = this.getMultilineSource(cell.source);
 		this._metadata = cell.metadata || {};
 
-		if (this._metadata.tags && this._metadata.tags.some(x => x === HideInputTag || x === ParametersTag || x === InjectedParametersTag) && this._cellType === CellTypes.Code) {
+		if (this._metadata.tags && this._metadata.tags.some(x => x === HideInputTag) && this._cellType === CellTypes.Code) {
 			this._isCollapsed = true;
+		} else if (this._metadata.tags && this._metadata.tags.some(x => x === ParametersTag) && this._cellType === CellTypes.Code) {
```

Comment:
So if a cell has multiple tags (hide input and parameters for example) you're only processing one. Is that what you intend?

I'd rewrite it something like this : 

```
if(this._metadata.tags && this._cellType === CellTypes.Code) { 
   const hideInputTag = this._metadata.tags.find(x => x === HideInputTag);
   this._isParameter = this._metadata.tags.find(x => x === ParametersTag);
   this._isInjectedParameter = this._metadata.tags.find(x => x === InjectedParametersTag);
   this._isCollapsed = hideInputTag || this._isParameter || this._isInjectedParameter;
} else {
   this._isCollapsed = false;
}
```

(you could also just use a single foreach over the tags array - but I expect that'll always be small enough to not really matter perf wise)

Reply:
Updated it. Small change on the fact that this._isCollapsed should only be looking at the hideInputTag, and not depend on isParameter and isInjectedParameter.